### ISSUE TITLE
Event will-download from DownloadItem only fires once

### ISSUE THREAD

**Issue by CGReinhold** [electron/electron#10673]
* Electron version: 1.7.6
* Operating system: Windows 10

### Expected behavior

Every time some the user clicks a button that downloads a file in the browser the event "will-download" should be fired.

### Actual behavior

Behavior 1. The event "will-download" is fired only for the first time some file is downloaded from the browser. If I click again to download a file the event is not fired anymore. 

Behavior 2. If I close and open the browser again (even without ending the application) the event if fired more than once (two, three, fore times) when I click in a button for download. 

### How to reproduce

I have this code in my project where I open a window in which the user can search and download some stuff:

```
let myBrowser = new BrowserWindow({ width: 1300, height: 1000, webPreferences: { nodeIntegration: false } })
myBrowser.setMenu(null)
myBrowser.loadURL('http://localhost/myurl.html')
var ses = myBrowser.webContents.session
ses.on('will-download', (event, item, webContents) => {
	var fileName = item.getFilename()
	var url = item.getURL()
	item.setSavePath(store.get('myFolder') + '/' + fileName)
	//DO SOME STUFF WITH THE FILE NAME
})
```

I don't prevent the download from beeing downloaded because all of what I need to do is save in a document the name and url of the file, and set the path to save the file.

**Comment by CGReinhold** [comment]
I also noticed that for the files that the download do work it's always downloaded in chunks of 128 bytes. So, for example, for files with less than 128 bytes it will create a file full of whitespace at the end to fit the size.

**Comment by CGReinhold** [comment]
I found the problem.

The Behavior 1 was due to a bug in the code of the page in which i was downloading the files.

The Behavior 2 was happening when the window was destroyed and then created again (same line of code). The second window would have the events duplicated, and then if you do it again the third window would have the event triplicated. I fixed it by keeping the first window all the time and only hiding and showing it.
